src/packet.cpp

```diff
@@ -0,0 +1,69 @@
+#include "packet.h"
+#include "packet_codes.h"
+#include <string>
+#include <iostream>
+#include <openssl/md5.h>
+
+Packet::Packet(const std::array<uint8_t, 4096>& m_recvBuffer)
+{
+    m_type = m_recvBuffer[0];
+
+    m_id = m_recvBuffer[1];
+
+    for (std::size_t i = 0; i < 16; ++i)
+        m_auth[i] = m_recvBuffer[i + 4];
+}
+
+uint8_t Packet::type() const
+{
+    return m_type;
+}
+
+uint8_t Packet::id() const
+{
+    return m_id;
+}
+
+const std::array<uint8_t, 16>& Packet::auth() const
+{
+    return m_auth;
+}
+
+std::vector<uint8_t>& Packet::makeSendBuffer()
+{
+    m_sendBuffer.resize(20);
```

Comment:
`m_sendBuffer` should be a local variable. And thus, the return type must be a value, not a reference.